xcube/core/gen2/local/opener.py

```diff
@@ -59,41 +62,47 @@ def open_cube(self, input_config: InputConfig) -> TransformedCube:
         open_params = input_config.open_params or {}
 
         with observe_progress('reading cube', 3) as observer:
-
-            if input_config.store_id:
-                store_instance = get_data_store_instance(
-                    input_config.store_id,
-                    store_params=store_params,
-                    store_pool=self._store_pool
-                )
-                store = store_instance.store
-                if opener_id is None:
-                    opener_id = self._get_opener_id(
-                        input_config, store
+            try:
+                if input_config.store_id:
+                    store_instance = get_data_store_instance(
+                        input_config.store_id,
+                        store_params=store_params,
+                        store_pool=self._store_pool
                     )
-                opener = store
-                open_params = dict(open_params)
-                open_params['opener_id'] = opener_id
-            else:
-                opener = new_data_opener(opener_id)
-                open_params = dict(open_params)
-                open_params.update(store_params)
-
-            open_params_schema = opener.get_open_data_params_schema(
-                input_config.data_id
-            )
-
-            dataset_open_params = {
-                k: v for k, v in cube_params.items()
-                if k in open_params_schema.properties
-            }
+                    store = store_instance.store
+                    if opener_id is None:
+                        opener_id = self._get_opener_id(
+                            input_config, store
+                        )
+                    opener = store
+                    open_params = dict(open_params)
+                    open_params['opener_id'] = opener_id
+                else:
+                    opener = new_data_opener(opener_id)
+                    open_params = dict(open_params)
+                    open_params.update(store_params)
+
+                open_params_schema = opener.get_open_data_params_schema(
+                    input_config.data_id
+                )
 
-            observer.worked(1)
+                dataset_open_params = {
+                    k: v for k, v in cube_params.items()
+                    if k in open_params_schema.properties
+                }
```

Comment:
I do not understand any of the changes above. What's the reason for this?

Reply:
This is all about catching DataStoreErrors. As there are several instances where one could be raised, I wrapped several lines of a code in a try block. You could think about moving these lines to a dedicated method, however, I did not want to do this as it is also wrapped in a with-block.